jira.py

```diff
@@ -32,6 +38,82 @@ def __init__(self):
     # API: Filters
     def filters(self):
         query = JQL_QUERY + FILTER_ID_ALL_REQUESTS_2022 + '&fields=' \
-                + DEFAULT_COLUMNS + STORY_POINTS + FIREFOX_RELEASE_TRAIN \
+                + DEFAULT_COLUMNS + ',' + STORY_POINTS + ',' \
+                + FIREFOX_RELEASE_TRAIN + ',' \
                 + ENGINEERING_TEAM + '&' + MAX_RESULT
+
         return self.client.get_search(query)
+
+
+class JiraClient(Jira):
+    def __init__(self):
+        super().__init__()
+        self.db = DatabaseJira()
+
+    def jira_qa_requests(self):
+        payload = self.filters()
+
+        data_frame = self.db.report_jira_qa_requests_payload(payload)
+        print(data_frame)
+
+        self.db.report_jira_qa_requests_insert(data_frame)
+
+
+class DatabaseJira(Database):
+
+    def __init__(self):
+        super().__init__()
+        self.db = Database()
+
+    def report_jira_qa_requests_payload(self, payload):
+        # Normalize the JSON data
+        df = pd.json_normalize(payload, sep='_')
+
+        # Check if 'jira_assignee_username' exists
+        # if not use 'alternative_assignee_emailAddress'
+        if 'fields_assignee_emailAddress' not in df.columns:
+            df['fields_assignee_emailAddress'] = df.get('fields_assignee', "None") # noqa
+        else:
+            df['fields_assignee_emailAddress'] = df['fields_assignee_emailAddress'].fillna("Not Assigned") # noqa
+
+        # Drop the alternative column if it exists
+        if 'fields_assignee' in df.columns:
+            df.drop(columns=['fields_assignee'], inplace=True)
+
+        # Select specific columns
+        selected_columns = {
+            'key': 'jira_key',
+            'fields_summary': 'jira_summary',
+            'fields_created': 'jira_created_at',
+            'fields_customfield_10155_value': 'jira_firefox_release_train',
+            'fields_customfield_10134_value': 'jira_engineering_team',
+            'fields_customfield_10037': 'jira_story_points',
+            'fields_status_name': 'jira_status',
+            'fields_assignee_emailAddress': 'jira_assignee_username',
+            'fields_labels': 'jira_labels'
+        }
+
+        # Select specific columns
+        df_selected = df[selected_columns.keys()]
+
+        # Rename columns
+        df_selected = df_selected.rename(columns=selected_columns)
+
+        df_selected['jira_created_at'] = df_selected['jira_created_at'].apply(dt.convert_to_utc) # noqa
+
+        # Join list of labels into a single string
+        df_selected['jira_labels'] = df_selected['jira_labels'].apply(lambda x: ','.join(x) if isinstance(x, list) else x) # noqa
+
+        return df_selected
+
+    def report_jira_qa_requests_insert(self, payload):
+        pass
+        # TBD
+        # This is the way used in testrail.py
+        report = ReportJiraQARequests()
```

Comment:
I think I have to especify the columns here so that this works, but I can't get it working with some changes I'm doing locally @rpappalax I will schedule sometime for this..

Reply:
Finally got how it works... Very close to have this working 🤞